### PR TITLE
Fix autoupdate with maintenance windows

### PR DESCRIPTION
**Pull request by parlt91**
## What does this PR change?

Make the auto errata update respect maintenance windows
Fix a bug related to auto errata update when a CLM build of affected channel is in progress.

## GUI diff

No difference.

- [x] **DONE**

## Documentation

- No documentation needed: **Bugfix**

- [x] **DONE**

## Test coverage

- No tests: **Bugfix**

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/15720

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by parlt91** [comment]
@mcalmer Could you have a look at this PR? It contains the changes for the maintenance windows we discussed a couple weeks ago

File: java/code/src/com/redhat/rhn/taskomatic/task/AutoErrataTask.java [hunk 1]
```diff
@@ -121,10 +162,16 @@ public void execute(JobExecutionContext context)
          * where rtb.name = 'repo-sync-bunch'
          *   and rtr.status in ('RUNNING','READY')
          */
+        if (sids.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        Map<String, String> params = new HashMap<>();
+        params.put("system_ids", "{" + sids + "}");
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/taskomatic/task/AutoErrataTask.java:170]: I think this does not look good. Check `systemCurrencyList()` for an alternative way.

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/Task_queries.xml [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 select distinct s.id as server_id, s.org_id, snc.errata_id
   from rhnServer s,
-       rhnServerFeaturesView sfv,
        rhnServerNeededCache snc
- where upper(s.auto_update) = 'Y'
-   and s.id = sfv.server_id
-   and sfv.label = 'ftr_auto_errata_updates'
-   and s.id = snc.server_id
+   where s.id = snc.server_id
+   and s.id in (select unnest(cast(:system_ids as int[])))
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/db/datasource/xml/Task_queries.xml:20]: Check my other comment for a different way of dealing with this.

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/Task_queries.xml (hunk 2 above)
Comment (parlt91) [reply java/code/src/com/redhat/rhn/common/db/datasource/xml/Task_queries.xml]: Indeed this is a lot better. Thanks for pointing it out.

**Review by mcalmer — APPROVED** [review]
Let's give it a try

File: java/code/src/com/redhat/rhn/taskomatic/task/AutoErrataTask.java [hunk 3]
```diff
@@ -94,6 +106,33 @@ public void execute(JobExecutionContext context)
...
+     *
+     * @param systems systems with auto errata enabled
+     * @return list of systems in maintenance mode
+     */
+    protected List<Long> filterSystemsInMaintenanceMode(List<Map<String, Long>> systems) {
+        MaintenanceManager mm = new MaintenanceManager();
+        return systems.stream()
+                .filter(system -> mm.isSystemInMaintenanceMode(ServerFactory.lookupById(system.get("id"))))
+                .map(system -> system.get("id"))
+                .collect(Collectors.toList());
```
Comment (cbbayburt) [inline java/code/src/com/redhat/rhn/taskomatic/task/AutoErrataTask.java:133]: I'm not sure how efficiently Hibernate will handle this, but maybe querying all IDs at once is a better choice.
See: `ServerFactory.lookupByIds`

**Review by cbbayburt — APPROVED** [review]
LGTM :+1: 

Just a couple of minor things below

**Pull request merged**